client_plugin/drivers/shared/shared_driver.go

```diff
@@ -23,24 +23,90 @@ package shared
 ///
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
 	log "github.com/Sirupsen/logrus"
 	dockerClient "github.com/docker/engine-api/client"
+	dockerTypes "github.com/docker/engine-api/types"
 	"github.com/docker/go-plugins-helpers/volume"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/drivers/utils"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/utils/config"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/utils/refcount"
+	"os"
+	"strconv"
 )
 
+// volStatus: Datatype for keeping status of a shared volume
+type volStatus string
+
+/* Constants
+   version:             Version of the shared plugin driver
+   apiVersion:          docker engine 1.24 and above support this api version
+   dockerUSocket:       Unix socket on which Docker engine is listening
+   tradVolumePrefix:    Prefix for the names of internal volumes. These
+                        volumes server as backend stores for shared volumes.
+   sambaImageName:      Name of the docker hub image we pull as samba server
+   sambaUsername:       Default username for all accessing Samba servers
+   sambaPassword:       Default password for accessing all Samba servers
+
+   volStateCreating:    Shared volume is being created. Not ready to be mounted.
+   volStateReady:       Shared volume is ready to be mounted but
+                        no Samba service running right now.
+   volStateMounted:     Samba service already running. Volume mounted
+                        on at least one host VM.
+   volStateIntermediate: Metadata of shared volume is being changed.
+                         Dont proceed with your op if this is the status.
+*/
 const (
-	version       = "vSphere Shared Volume Driver v0.2"
-	apiVersion    = "v1.24" // docker engine 1.24 and above support this api version
-	dockerUSocket = "unix:///var/run/docker.sock"
+	version          = "vSphere Shared Volume Driver v0.2"
+	apiVersion       = "v1.24"
+	dockerUSocket    = "unix:///var/run/docker.sock"
+	tradVolumePrefix = "InternalTrad"
+	// TODO Replace with our own samba later
+	sambaImageName                 = "dperson/samba"
+	sambaUsername                  = "root"
+	sambaPassword                  = "badpass"
+	volStateCreating     volStatus = "Creating"
+	volStateReady        volStatus = "Ready"
+	volStateMounted      volStatus = "Mounted"
+	volStateIntermediate volStatus = "Intermediate"
```

Comment:
Intermediate was supposed to be a state that signals other jobs not to proceed. The plan was to use this state for any meta data update, including delete.

Changing its text to "MetadataUpdateInProgress". Will introduce Error state in another PR.